### PR TITLE
[Bugfix:TAGrading]Peer grading team interface

### PR DESCRIPTION
**Pull request by jchulton**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Peer Grading cannot be done for team assignments. Creating a peer mark with a team assignment leads to untested behavior.

### What is the new behavior?
Peer grading can now be done for team assignments. Peer marks are created as normal. Students are assigned to grade another students, just like normal peer assignments. A peer can grade the teams which contains the students they are assigned to grade. Peer grading of teams functions the same way as peer grading of individuals.

### Other information?
 No
 Walking through peer grading via teams

This should be merged after peer grading (PR #4970)

### CODE REVIEW

**Comment by bmcutler** [comment]
@jchulton or @emaicus please resolve conflicts

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 1]
```diff
@@ -1672,9 +1678,9 @@ public function saveComponentsPage(Gradeable $gradeable, int $page) {
 
 /**
      * Route for adding a new component to a gradeable
-     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/components/new", methods={"POST"})
+     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/components/{peer}/new", methods={"POST"})
```
Comment (emaicus) [inline site/app/controllers/grading/ElectronicGraderController.php:1681]: Remove peer from the route (it's not there in master)

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 2]
```diff
@@ -1672,9 +1678,9 @@ public function saveComponentsPage(Gradeable $gradeable, int $page) {
...
      */
-    public function ajaxAddComponent($gradeable_id) {
+    public function ajaxAddComponent($gradeable_id, $peer) {
```
Comment (emaicus) [inline site/app/controllers/grading/ElectronicGraderController.php:1683]: same as above. Peer should be passed in ```$POST``` now.

File: site/app/templates/admin/users/GraderList.twig [hunk 3]
```diff
@@ -6,6 +6,7 @@
             <a href="javascript:newGraderListForm()" class="btn btn-primary">Upload Grader List</a>
             <a href="javascript:newGraderForm()" class="btn btn-primary">New Grader</a>
             <a href="javascript:editRegistrationSectionsForm()" class="btn btn-primary">Assign Registration Sections for Grading</a>
+            <div class="btn" onclick="uploadPeerResults('test2', '{{ csrf_token }}');">Peer stuff</div>
```
Comment (emaicus) [inline site/app/templates/admin/users/GraderList.twig:9]: Remove this debug code.

File: site/app/templates/grading/electronic/Status.twig [hunk 4]
```diff
@@ -627,12 +627,12 @@
                         {% endif %}
 
                         {% if ta_grades_released %}
-                            {% if team_assignment %}
+                            {% if team_assignment and not peer %}
```
Comment (emaicus) [inline site/app/templates/grading/electronic/Status.twig:630]: There is no peer AND team clause in this if statement. Should there be?

File: site/public/js/ta-grading-rubric.js [hunk 5]
```diff
@@ -626,7 +626,7 @@ function ajaxAddComponent(gradeable_id, peer) {
         $.getJSON({
             type: "POST",
             async: AJAX_USE_ASYNC,
-            url: buildCourseUrl(['gradeable', gradeable_id, 'components', 'new']),
+            url: buildCourseUrl(['gradeable', gradeable_id, 'components', peer, 'new']),
```
Comment (emaicus) [inline site/public/js/ta-grading-rubric.js:629]: Again, remove peer from the route. It should be passed in the POST

File: site/public/templates/grading/EditComponent.twig [hunk 6]
```diff
@@ -110,6 +110,23 @@
                 </span>
             </div>
         </div>
+       {# <div class="row point-input-row peer-type-row">
```
Comment (emaicus) [inline site/public/templates/grading/EditComponent.twig:113]: Remove this

File: site/public/templates/grading/GradingComponent.twig [hunk 7]
```diff
@@ -27,6 +27,11 @@ Required inputs:
     <div class="container">
         {% set show_ta_comment = component.ta_comment != "" %}
         {% set show_student_comment = component.student_comment != "" %}
+        {% if component.peer %}
```
Comment (emaicus) [inline site/public/templates/grading/GradingComponent.twig:30]: Remove this

**Comment by emaicus** [comment]
I got the following javascript error when uploading a peer mapping csv for a team assignment:
```
Error parsing twig template /templates/grading/GradingComponentHeader.twig: 
TypeError: Cannot read property 'options' of undefined
    at twig.min.js?v=1586974800:9
    at p.Thenable.resolvedThen [as then] (twig.min.js?v=1586974800:9)
    at parse (twig.min.js?v=1586974800:9)
    at twig.min.js?v=1586974800:9
    at e (twig.min.js?v=1586974800:9)
    at Object.p.async.forEach (twig.min.js?v=1586974800:9)
    at twig.min.js?v=1586974800:9
    at Object.p.async.potentiallyAsync (twig.min.js?v=1586974800:9)
    at u.expression.parse (twig.min.js?v=1586974800:9)
    at p.expression.parseAsync (twig.min.js?v=1586974800:9)
```

**Comment by emaicus** [comment]
Students cannot see the "names" of the teams that they are going to be grading (anonymous or otherwise).

**Comment by emaicus** [comment]
Grading teams does not properly fill in the "bubbles" for the corresponding questions on the grading index page.

**Comment by jchulton** [comment]
The javascript error is on master on the Grader Assignment page. It doesn't seem to cause any problems on master or this branch

**Pull request merged**